File: C/Easy/1346.c
bool checkIfExist(int *arr, int arrSize)
{
    // use a hashtable to store seen numbers
    int *table = calloc(2001, sizeof(int));
    int num;
    for (int i = 0; i < arrSize; i++)
    {
        num = arr[i];

        // check if exist the divison and multiple
        if (num % 2 == 0)
        {
            if (num / 2 + 1000 >= 0 && num / 2 + 1000 < 2001)
            {
                if (table[num / 2 + 1000])
                {
                    return true;
                }
            }
        }

        if (num * 2 + 1000 >= 0 && num * 2 + 1000 < 2001)
        {
            if (table[num * 2 + 1000])
            {
                return true;
            }
        }

        num = arr[i] + 1000;

        table[num] = 1;
    }

    return false;
}

File: Python/Easy/1346.py
class Solution {
    public boolean checkIfExist(int[] arr) {
        // store a set of the numbers seen so far
        Set<Integer> set = new HashSet<>();
        for (int i : arr) {
            if (set.contains(i * 2) || (i % 2 == 0 && set.contains(i / 2))) {
                return true;
            }
            set.add(i);
        }

        return false;
    }
}